src/crypto/asset_proofs.cpp

```
@@ -0,0 +1,952 @@
+// as the auxiliary commitment's value-base, so the unmodified Bulletproof+
+// engine can be reused without modification.
+
+static rct::key hash_buffer_to_scalar(const std::vector<uint8_t>& buf)
+{
+    rct::key result;
+    keccak(buf.data(), (int)buf.size(), result.bytes, 32);
+    sc_reduce32(result.bytes);
+    return result;
+}
+
+bool generate_vector_ug_aggregation_proof(const rct::key&  context_hash,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)